pennylane/qchem/integrals.py

```diff
@@ -138,14 +138,29 @@ def _generate_params(params, args):
     Returns:
         list(array[float]): basis set parameters
     """
+    # Friendly reference for JAX.
+    # params order is ALPHA COEFF COORD
+    # args order is COORD COEFF ALPHA
     basis_params = []
     c = 0
-    for p in params:
-        if p.requires_grad:
-            basis_params.append(args[c])
-            c += 1
-        else:
+    for i, p in enumerate(params):
+        if len(args) == 0:
             basis_params.append(p)
+            continue
+
+        interface = qml.math.get_interface(p)
+        if interface == "autograd":
+            if getattr(p, "requires_grad", False):
+                basis_params.append(args[c])
+                c += 1
+            else:
+                basis_params.append(p)
+        else:
```

Comment:
`else` might also contain `numpy` interface, is that intended?

Reply:
Yes. Since only `autograd` has the `requires_grad` attr, all other interfaces should go through that `else` statement.